File: backend/src/models/Schedule.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Schedule = sequelize.define('Schedule', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    therapist_id: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'therapists',
        key: 'id',
      },
    },
    date: {
      type: DataTypes.DATEONLY,
      allowNull: false,
      validate: {
        isDate: { msg: 'Invalid date format' },
      },
    },
    start_time: {
      type: DataTypes.TIME,
      allowNull: false,
    },
    end_time: {
      type: DataTypes.TIME,
      allowNull: false,
    },
    is_booked: {
      type: DataTypes.BOOLEAN,
      defaultValue: false,
    },
  }, {
    tableName: 'schedules',
    timestamps: true,
    underscored: true,
    indexes: [
      { fields: ['therapist_id'] },
      { fields: ['date'] },
      { fields: ['is_booked'] },
      { fields: ['therapist_id', 'date'] },
    ],
    validate: {
      endTimeAfterStartTime() {
        if (this.start_time >= this.end_time) {
          throw new Error('End time must be after start time');
        }
      },
    },
  });

  return Schedule;
};

File: mobile/lib/models/schedule.dart
class Schedule {
  final String id;
  final String therapistId;
  final String date;
  final String startTime;
  final String endTime;
  final bool isBooked;

  Schedule({
    required this.id,
    required this.therapistId,
    required this.date,
    required this.startTime,
    required this.endTime,
    this.isBooked = false,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'] ?? '',
      therapistId: json['therapist_id'] ?? '',
      date: json['date'] ?? '',
      startTime: json['start_time'] ?? '',
      endTime: json['end_time'] ?? '',
      isBooked: json['is_booked'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'therapist_id': therapistId,
      'date': date,
      'start_time': startTime,
      'end_time': endTime,
      'is_booked': isBooked,
    };
  }
}
